compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpInterceptorServiceValidator.java

```diff
@@ -83,34 +99,223 @@ private static void validateInterceptorTypeAndProceed(SyntaxNodeAnalysisContext
+                        reportInvalidParameterType(ctx, paramLocation, paramType);
+                        continue;
+                    }
+                    Optional<String> nameSymbolOptional = moduleSymbolOptional.get().getName();
+                    if (nameSymbolOptional.isEmpty()) {
+                        reportInvalidParameterType(ctx, paramLocation, paramType);
+                        continue;
+                    }
+                    if (!HTTP.equals(nameSymbolOptional.get())) {
+                        reportInvalidParameterType(ctx, paramLocation, paramType);
+                        continue;
+                    }
+
+                    Optional<String> typeNameOptional = typeDescriptor.getName();
+                    if (typeNameOptional.isEmpty()) {
+                        reportInvalidParameterType(ctx, paramLocation, paramType);
+                        continue;
+                    }
+                    switch (typeNameOptional.get()) {
+                        case CALLER_OBJ_NAME:
+                            callerPresent = isObjectPresent(ctx, paramLocation, callerPresent, paramName,
+                                    HttpDiagnosticCodes.HTTP_115);
+                            break;
+                        case RESPONSE_OBJ_NAME:
+                            responsePresent = isObjectPresent(ctx, paramLocation, responsePresent, paramName,
+                                    HttpDiagnosticCodes.HTTP_139);
+                            break;
+                        case REQUEST_CONTEXT_OBJ_NAME:
+                            requestCtxPresent = isObjectPresent(ctx, paramLocation, requestCtxPresent, paramName,
+                                    HttpDiagnosticCodes.HTTP_121);
+                            break;
+                        default:
+                            reportInvalidParameterType(ctx, paramLocation, paramType);
+                            break;
+                    }
+                } else {
+                    reportInvalidParameterType(ctx, paramLocation, paramType);
+                }
+            } else if (kind == TypeDescKind.ERROR) {
+                if (isResponseErrorInterceptor(type)) {
+                    errorPresent = isObjectPresent(ctx, paramLocation, errorPresent, paramName,
+                            HttpDiagnosticCodes.HTTP_122);
+                } else {
+                    reportInvalidParameterType(ctx, paramLocation, paramType);
+                }
+            } else {
+                reportInvalidParameterType(ctx, paramLocation, paramType);
+            }
+            continue;
         }
     }
 
+    private static boolean isObjectPresent(SyntaxNodeAnalysisContext ctx, Location location,
+                                           boolean objectPresent, String paramName, HttpDiagnosticCodes code) {
+        if (objectPresent) {
+            HttpCompilerPluginUtil.updateDiagnostic(ctx, location, code, paramName);
+        }
+        return true;
```

Comment:
This is to check for multiple same parameters. This will check whether `objectPresent` is true and will create an error diagnostic. Method will always returns `true` and this value is set again for the boolean value which checks the existence of a parameter.

```java
switch (typeNameOptional.get()) {
   case CALLER_OBJ_NAME:
       callerPresent = isObjectPresent(ctx, paramLocation, callerPresent, paramName, HTTP_115);
       break;
```

Same as : 

```java
switch (typeNameOptional.get()) {
   case CALLER_OBJ_NAME:
       if (callerPresent) {
           // report error diagnostic
       }
       callerPresent = true;
       break;
```

This was already there for resource parameter validation, I have just used the same logic for interceptor remote function